### PR TITLE
OLS-86: Overall service architecture

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Overall service architecture

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [x] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)
- [ ] CI configuration change
- [ ] Konflux configuration change

## Related Tickets & Documents

- Related Issue #OLS-86

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -568,10 +568,61 @@ Chart customization is available using the [Values](helm/values.yaml) file.
...
 
+OpenShift LightSpeed service is based on FastAPI framework (Uvicorn) with Langchain for LLM interactions. The service is split into several parts described below.
+
+### FastAPI server
+
+Handles REST API requests from clients (mainly from UI console, but can be any REST API-compatible tool), handles requests queue, and also exports Prometheus metrics. Uvicorn framework is used as FastAPI implementation.
+
+### Authorization checker
+
+Manages authentication flow for REST API endpoints. Currently K8S/OCL-based authorization is used, but in future it will be implemented in a more modular way to allow registering other auth. checkers.
```
Comment (JoaoFula) [inline README.md:583]: but in **the** future it will be implemented...

File: README.md [hunk 2]
```diff
@@ -568,10 +568,61 @@ Chart customization is available using the [Values](helm/values.yaml) file.
...
+
+### Query handler
+
+Retrieves user queries, validates them, redact them, call LLM, and summarize feedback.
```
Comment (JoaoFula) [inline README.md:587]: redacts (...) calls (...) summarizes

File: README.md [hunk 3]
```diff
@@ -568,10 +568,61 @@ Chart customization is available using the [Values](helm/values.yaml) file.
...
+1. Redis cache
+1. Postgres cache
+
+### LLM providers registry
+
+Manages LLM providers implementations. If new LLM provider type needs to be added, it is registered by this machinery and its libraries are loaded to be used later.
+
+### LLM providers interface implementations
+
+Currently there exist following LLM providers implementations:
```
Comment (JoaoFula) [inline README.md:618]: ... there exist **the** following...

File: README.md (hunk 1 above)
Comment (tisnik) [reply README.md]: fixed, thank you

File: README.md (hunk 2 above)
Comment (tisnik) [reply README.md]: fixed, thank you

File: README.md (hunk 3 above)
Comment (tisnik) [reply README.md]: fixed, thank you

**Comment by tisnik** [comment]
/override "Red Hat Konflux / lightspeed-service-on-pull-request"

**Comment by raptorsun** [comment]
fix for E2E test https://github.com/openshift/lightspeed-service/pull/1659

**Pull request merged**